labs/authentication-passport/src/strategy-adapter.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright IBM Corp. 2017,2019. All Rights Reserved.
+// Node module: @loopback/authentication-passport
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {AuthenticationStrategy, UserProfile} from '@loopback/authentication';
+import {HttpErrors, Request} from '@loopback/rest';
+import {Strategy} from 'passport';
+
+const passportRequestMixin = require('passport/lib/http/request');
+
+/**
+ * Adapter class to invoke passport-strategy
+ *   1. provides express dependencies to the passport strategies
+ *   2. provides shimming of requests for passport authentication
+ *   3. provides lifecycle similar to express to the passport-strategy
+ *   4. provides state methods to the strategy instance
+ * see: https://github.com/jaredhanson/passport
+ */
+export class StrategyAdapter implements AuthenticationStrategy {
+  passportStrategy: Strategy;
+  /**
+   * @param strategy instance of a class which implements a passport-strategy;
+   * @description http://passportjs.org/
+   */
+  constructor(private readonly strategy: Strategy, readonly name: string) {
+    this.name = name;
+    this.passportStrategy = strategy;
+  }
+
+  /**
+   * The function to invoke the contained passport strategy.
+   *     1. Create an instance of the strategy
+   *     2. add success and failure state handlers
+   *     3. authenticate using the strategy
+   * @param request The incoming request.
+   */
+  authenticate(request: Request): Promise<UserProfile> {
+    return new Promise<UserProfile>((resolve, reject) => {
+      // mix-in passport additions like req.logIn and req.logOut
+      for (const key in passportRequestMixin) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (request as any)[key] = passportRequestMixin[key];
+      }
+
+      // create a prototype chain of an instance of a passport strategy
+      const strategy = Object.create(this.strategy);
```

Comment:
Maybe I missed something, but wouldn't this be `Object.create(this.passportStrategy);`?

Reply:
Good catch `passportStrategy` is a dup. Removed.